File: backend/mallapi/src/main/java/org/nareun130/mallapi/controller/ProductController.java
package org.nareun130.mallapi.controller;

import java.util.List;
import java.util.Map;

import org.nareun130.mallapi.dto.ProductDTO;
import org.nareun130.mallapi.util.CustomFileUtil;
import org.springframework.core.io.Resource;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.multipart.MultipartFile;

import lombok.RequiredArgsConstructor;
import lombok.extern.log4j.Log4j2;

@RestController
@RequiredArgsConstructor
@Log4j2
@RequestMapping("/api/products")
public class ProductController {
    
    private final CustomFileUtil fileUtil;

    @PostMapping("/")
    public Map<String, String> register(ProductDTO productDTO){
        
        log.info("register: " + productDTO); 

        List<MultipartFile> files = productDTO.getFiles();

        List<String> uploadFileNames = fileUtil.saveFiles(files);

        productDTO.setUploadFileNames(uploadFileNames);

        log.info(uploadFileNames);

        return Map.of("RESULT", "SUCCESS");
    }

    @GetMapping("/view/{fileName}")
    public ResponseEntity<Resource> viewFileGET(@PathVariable("fileName") String fileName) {
        return fileUtil.getFile(fileName);
    }

}

File: backend/mallapi-kotlin/src/main/kotlin/com/nareun/mallapikotlin/controller/ProductController.kt
package com.nareun.mallapikotlin.controller

import com.nareun.mallapikotlin.dto.PageRequestDTO
import com.nareun.mallapikotlin.dto.PageResponseDTO
import com.nareun.mallapikotlin.dto.ProductDTO
import com.nareun.mallapikotlin.service.ProductService
import com.nareun.mallapikotlin.util.CustomFileUtil
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.core.io.Resource
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import org.springframework.web.multipart.MultipartFile

@RestController
@RequestMapping("/api/products")
class ProductController(
    private val productService: ProductService,
    private val fileUtil: CustomFileUtil
) {

    val log = LoggerFactory.getLogger(ProductController::class.java)

    @GetMapping("/list")
    fun list(pageRequestDTO: PageRequestDTO):PageResponseDTO<ProductDTO>{

        log.info("list.........$pageRequestDTO")

        return productService.getList(pageRequestDTO)
    }

    @PostMapping("/")
    fun register(productDTO: ProductDTO): Map<String, String> {
        log.info("register: $productDTO")

        val files: List<MultipartFile>? = productDTO.files

        val uploadFileNames = fileUtil.saveFiles(files)

        productDTO.uploadFileNames = uploadFileNames

        log.info(uploadFileNames.toString())

        return mutableMapOf("RESULT" to "SUCCESS")
    }

    @GetMapping("/view/{fileName}")
    fun viewFileGet(@PathVariable("fileName") fileName: String): ResponseEntity<Resource> {
        return fileUtil.getFile(fileName)
    }
}